From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Josh Gunter <59844337+jagunter@users.noreply.github.com>
Date: Tue, 28 Mar 2023 11:32:57 -0700
Subject: [PATCH] 5176: Throw exception on invalid address space element during
 decode

Throw exception if unexpected element encountered while decoding address space from .sla spec
---
 Ghidra/Features/Decompiler/src/decompile/cpp/translate.cc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/translate.cc b/Ghidra/Features/Decompiler/src/decompile/cpp/translate.cc
index 58aa033f3a..df09a27bf8 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/translate.cc
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/translate.cc
@@ -224,8 +224,10 @@ AddrSpace *AddrSpaceManager::decodeSpace(Decoder &decoder,const Translate *trans
     res = new OtherSpace(this,trans);
   else if (elemId == ELEM_SPACE_OVERLAY)
     res = new OverlaySpace(this,trans);
-  else
+  else if (elemId == ELEM_SPACE)
     res = new AddrSpace(this,trans,IPTR_PROCESSOR);
+  else
+    throw LowlevelError("Invalid address space element. Internal element id: " + to_string(elemId));
 
   res->decode(decoder);
   return res;
-- 
2.45.0

